### PR TITLE
Added password confirmation to register page

### LINKED ISSUE #94: Separate Login and Register page
Login Page:
* username:
* password:
* login button

Register Page:
* username:
* password:
* confirm password:
* register button

### PR DESCRIPTION
**Pull request by tnttruong**
image

Register page now has a `Confirm password` field which checks that the string inputted in `Password` is the same as the one inputted in `Confirm password`. This PR also updates all unit tests to include fields for confirm password, as well as a unit test that checks the passwords are identical.

Closes #94

### CODE REVIEW

**Comment by Brod8362** [comment]
Can you also fix #94 in this PR?

File: src/backend/src/database.rs [hunk 1]
```diff
@@ -184,11 +186,18 @@ fn is_username_valid(username: &str) -> bool {
     true
 }
 
+fn is_password_confirmed(password: &str, password_confirm: &str) -> bool {
+    match password == password_confirm {
+        true => return true,
+        _ => return false,
+    }
```
Comment (Brod8362) [inline src/backend/src/database.rs:193]: This entire statement can be simplified to `password ==  password_confirm`

File: src/backend/src/database.rs [hunk 2]
```diff
@@ -184,11 +186,18 @@ fn is_username_valid(username: &str) -> bool {
...
+}
+
 /// Register a new user in the database.
 /// This function will ensure the username is not duplicated. 
 /// It also handles hashing and salting the password.
 /// The ID of the created user will be returned.
-pub async fn register_user(pool: &Pool<Sqlite>, username: &str, password: &str) -> Result<String, Box<dyn Error>> {
+pub async fn register_user(pool: &Pool<Sqlite>, username: &str, password: &str, password_confirm: &str) -> Result<String, Box<dyn Error>> {
```
Comment (Brod8362) [inline src/backend/src/database.rs:200]: Deal with the confirmation in the caller (e.g the route), not here.

File: src/backend/templates/register_page.html.tera [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            {% endif %}
+
+            <br>
+            <div id="button-container">
+            <button type="submit" hx-get="/login">
+                <div class="form-sect">
+                    <img src="/static/checkmark_icon1.png" alt="arrow icon ">
+                    <span>Login</span>
+                </div>
+            </button>
```
Comment (Brod8362) [inline src/backend/templates/register_page.html.tera:57]: Should this button really be here on the register page? I'm not sure if that makes sense.
IMO, it should only be the register button

**Review by Brod8362 — CHANGES_REQUESTED** [review]
See comments

File: src/backend/src/database.rs (hunk 2 above)
Comment (Brod8362) [reply src/backend/src/database.rs:200]: This does mean you'll have to revert all of the changes in the unit tests

Comment (tnttruong) [reply src/backend/src/database.rs:200]: What do you mean by deal with the confirmation in the caller? Do you mean only checking for matching passwords in `routes.rs`?

Comment (Brod8362) [reply src/backend/src/database.rs:197]: Yes

**Comment by Brod8362** [comment]
Do note a rebase is required

File: src/backend/templates/register_page.html.tera (hunk 3 above)
Comment (tnttruong) [reply src/backend/templates/register_page.html.tera]: image

added a basic authentication page that now leaves each page with only 1 submit button

**Review by Mnguyener — COMMENTED** [review]
After registering, can you direct the user to the login page?

**Review by Brod8362 — CHANGES_REQUESTED** [review]
Looks good, but I agree the user should be redirect to the login page after registering.

**Pull request merged**